zarr/hierarchy.py

```diff
@@ -523,6 +523,34 @@ def visititems(self, func):
         base_len = len(self.name)
         return self.visitvalues(lambda o: func(o.name[base_len:].lstrip("/"), o))
 
+    def tree(self):
+        """Provide a ``print`-able display of the hierarchy.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> g1 = zarr.group()
+        >>> g2 = g1.create_group('foo')
+        >>> g3 = g1.create_group('bar')
+        >>> g4 = g3.create_group('baz')
+        >>> g5 = g3.create_group('quux')
+        >>> d1 = g5.create_dataset('baz', shape=100, chunks=10)
+        >>> print(g1.tree())
```

Comment:
Suggest just:

```
>>> g1.tree()
```

I.e., no need to call print().

Reply:
Not a strong thought by any means, but `print` does force `str` -> `repr`. Other circumstances (e.g. in a Jupyter Notebook) not using `print` would have a different effect. Though again not a strong thought happy to change if it is preferred without `print`.